lib/puma/server.rb

```diff
@@ -426,7 +426,7 @@ def process_client(client, buffer)
 
             check_for_more_data = @status == :run
 
-            if requests >= MAX_FAST_INLINE
+            if requests >= MAX_FAST_INLINE && MAX_FAST_INLINE.positive?
```

Comment:
e.g. server should have `@max_fast_inline` set by the options, and then that is read here. Don't check here to see if it's positive, check when you're setting it.